src/main/java/net/fabricmc/loader/transformer/accessWidener/AccessWidener.java

```diff
@@ -0,0 +1,355 @@
+			makeClassExtendable(entryTriple);
+			return access.makeExtendable();
+		case "mutable":
+			return access.makeMutable();
+		default:
+			throw new UnsupportedOperationException("Unknown access type:" + input);
+		}
+	}
+
+	private void makeClassAccessible(EntryTriple entryTriple) {
+		if (entryTriple == null) return;
+		classAccess.put(entryTriple.getOwner(), applyAccess("accessible", classAccess.getOrDefault(entryTriple.getOwner(), ClassAccess.DEFAULT), null));
+	}
+
+	private void makeClassExtendable(EntryTriple entryTriple) {
+		if (entryTriple == null) return;
+		classAccess.put(entryTriple.getOwner(), applyAccess("extendable", classAccess.getOrDefault(entryTriple.getOwner(), ClassAccess.DEFAULT), null));
+	}
+
+	public Access getClassAccess(String className) {
+		return classAccess.getOrDefault(className, ClassAccess.DEFAULT);
+	}
+
+	public Access getFieldAccess(EntryTriple entryTriple) {
+		return fieldAccess.getOrDefault(entryTriple, FieldAccess.DEFAULT);
+	}
+
+	public Access getMethodAccess(EntryTriple entryTriple) {
+		return methodAccess.getOrDefault(entryTriple, MethodAccess.DEFAULT);
+	}
+
+	public Set<String> getTargets() {
+		return classes;
+	}
+
+	private static int makePublic(int i) {
+		return (i & ~(Opcodes.ACC_PRIVATE | Opcodes.ACC_PROTECTED)) | Opcodes.ACC_PUBLIC;
+	}
+
+	private static int makeProtected(int i) {
+		if ((i & Opcodes.ACC_PUBLIC) != 0) {
+			//Return i if public
+			return i;
+		}
+
+		return (i & ~(Opcodes.ACC_PRIVATE)) | Opcodes.ACC_PROTECTED;
+	}
+
+	private static int makeFinalIfPrivate(int i) {
+		if ((i & Opcodes.ACC_PRIVATE) != 0) {
+			return i | Opcodes.ACC_FINAL;
+		}
+
+		return i;
+	}
+
+	private static int removeFinal(int i) {
+		return i & ~Opcodes.ACC_FINAL;
+	}
+
+	public interface Access {
+		Access makeAccessible();
+
+		Access makeExtendable();
+
+		Access makeMutable();
+
+		int apply(int i);
+	}
+
+	public enum ClassAccess implements Access {
+		DEFAULT(i -> i),
+		ACCESSIBLE(i -> makePublic(makeFinalIfPrivate(i))),
+		EXTENDABLE(i -> makePublic(removeFinal(i))),
+		ACCESSIBLE_EXTENDABLE(i -> makePublic(removeFinal(i)));
+
+		private final IntUnaryOperator operator;
+
+		ClassAccess(IntUnaryOperator operator) {
+			this.operator = operator;
+		}
+
+		@Override
+		public Access makeAccessible() {
+			if (this == EXTENDABLE || this == ACCESSIBLE_EXTENDABLE) {
+				return ACCESSIBLE_EXTENDABLE;
+			}
+
+			return ACCESSIBLE;
+		}
+
+		@Override
+		public Access makeExtendable() {
+			if (this == ACCESSIBLE || this == ACCESSIBLE_EXTENDABLE) {
+				return ACCESSIBLE_EXTENDABLE;
+			}
+
+			return EXTENDABLE;
+		}
+
+		@Override
+		public Access makeMutable() {
+			throw new UnsupportedOperationException("Classes cannot be made mutable");
+		}
+
+		@Override
+		public int apply(int i) {
+			return operator.applyAsInt(i);
+		}
+	}
+
+	public enum MethodAccess implements Access {
+		DEFAULT(i -> i),
+		ACCESSIBLE(i -> makePublic(makeFinalIfPrivate(i))), //Make public, add final if private
```

Comment:
How about method being accessible by subclass but not other outer classes?

Reply:
Im not 100% sure what you mean? I dont think its an issue due to it actaully using the bridge method.

Reply:
I think ill just remove `makeFinalIfPrivate` it sounds like its going to cause more issues than it will fix.